bal/bcnode.py

```diff
@@ -75,66 +73,57 @@ def isAvailable( self ):
             cmd += self.client
         return quietRun(cmd)
 
-
-    def call(self, command, silent= False, data=''):
+    def call(self, command, silent=False, data=''):
         """Call <client> <cargs> on node."""
         if self.cdir is not None:
-            self.cmd( 'cd ' + self.cdir )
+            self.cmd('cd ' + self.cdir)
         cmd = self.client
-        pathCheck( cmd )
+        pathCheck(cmd)
         if data:
-            method = '''POST -H "Content-Type: application/json" -d '{data}' '''.format(data = data)
+            method = '''POST -H "Content-Type: application/json" -d '{data}' '''.format(data=data)
         else:
             method = "GET"
 
         if self.cargs:
-            cmd += " " + self.cargs.format(command=command,
-                                           method=method,
-                                           name=self.name,
-                                           IP=self.IP(),
-                                           port=self.port,
-                                           cdir=self.cdir,
-                                           sdir=self.sdir)
+            a = self.cargs.format(command=command, method=method, 
+                                  name=self.name, IP=self.IP(), 
+                                  port=self.port, cdir=self.cdir,
+                                  sdir=self.sdir)
+            cmd += " " + 
```

Comment:
this could be problematic, you forgot to add "a" variable. also naming "a" is little bit ambiguous.